packages/dapp-high-roller/src/components/app-game/app-game.tsx

```diff
@@ -0,0 +1,113 @@
+import { Component, Prop, State } from "@stencil/core";
+import { RouterHistory } from "@stencil/router";
+
+import { GameState, PlayerType } from "../../enums/enums";
+
+const DARK_PATH = "./assets/images/dice/Dark/Dice-Dark-0";
+const LIGHT_PATH = "./assets/images/dice/Light/Dice-Light-0";
+
+@Component({
+  tag: "app-game",
+  styleUrl: "app-game.scss",
+  shadow: true
+})
+export class AppGame {
+  @Prop() history: RouterHistory;
+
+  @Prop({ mutable: true }) myName: string = "Facundo";
+  @Prop({ mutable: true }) betAmount: string = "3 ETH";
+  @Prop() opponentName: string = "John";
+
+  @State() gameState: GameState = GameState.Play;
+  @State() myRoll: number[] = [1, 1];
+  @State() myScore: number = 0;
+
+  @State() opponentRoll: number[] = [1, 1];
+  @State() opponentScore: number = 0;
+
+  componentWillLoad() {
+    this.myName = this.history.location.state.myName
+      ? this.history.location.state.myName
+      : this.myName;
+    this.betAmount = this.history.location.state.betAmount
+      ? this.history.location.state.betAmount
+      : this.betAmount;
+  }
+
+  handleRoll(): void {
+    this.myRoll = [
+      1 + Math.floor(Math.random() * Math.floor(6)),
```

Comment:
@Alonski We can probably abstract this into a `getRandomNumber()` function:

```ts
const getRandomNumber = (max) => 1 + Math.floor(Math.random() * max);
```

and then use it as:

```ts
this.myRoll = [getRandomNumber(6), getRandomNumber(6)]
```

Also you can go further and have a `getRoll()` function that builds the array for you.

Reply:
This is currently boilerplate code that will be thrown away once we integrate with a smart contract. I can do this though.

Reply:
Don't sweat it. Let's get this merged asap :)